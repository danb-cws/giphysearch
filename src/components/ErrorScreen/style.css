.Error-screen {
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  color: var(--secondary-color-inverse);
  background-color: var(--secondary-bg-color-inverse);
  font-size: 2em;
  padding: 1em;
  width: 100%;
  height: 100%;
}
