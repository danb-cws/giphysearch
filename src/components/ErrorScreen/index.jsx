import React from "react";
import "./style.css";

const ErrorScreen = props => {
  if (props.gifs.length === 0 && props.jsonIsLoaded) {
    return (
      <div className="ErrorScreen">
          {!props.searchTerm
            ? "Please enter a term in the search field"
            : `No gifs available for "${props.searchTerm}"`}
      </div>
    );
  } else {
    return null;
  }
};

export default ErrorScreen;
