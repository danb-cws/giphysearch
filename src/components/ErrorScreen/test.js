import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import ErrorScreen from "./";

configure({ adapter: new Adapter() });

describe("ErrorScreen tests", () => {

  it("Responds with 'please enter term' if no gifs and json has loaded", function() {
    expect(
      shallow(
        <ErrorScreen gifs={[]} searchTerm={""} jsonIsLoaded={true} />
      ).contains(
        <div className="ErrorScreen">
          Please enter a term in the search field
        </div>
      )
    ).toBe(true);
  });

  it("Responds with 'no matches for testTerm' if no gifs and json has loaded and searchTerm exists", () => {
    expect(
      shallow(
        <ErrorScreen gifs={[]} searchTerm={"testTerm"} jsonIsLoaded={true} />
      ).contains(
        <div className="ErrorScreen">No gifs available for "testTerm"</div>
      )
    ).toBe(true);
  });
});
