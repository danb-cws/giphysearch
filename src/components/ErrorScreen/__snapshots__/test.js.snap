// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorScreen tests Renders and matches snapshot 1`] = `
<div
  className="ErrorScreen"
>
  No gifs available for "test term"
</div>
`;
